app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    persons = []
+
+    for i in people_list:
+        name = i["name"]
+        age = i["age"]
+        person = Person(name, age)
+        persons.append(person)
+
+    for person in people_list:
+        if "wife" in person:
+            if person["wife"] is not None:
```

Comment:
you can use dict get() method to simplify this

Reply:
Actually if there's no key adressing the key of dict will throw an error, while method `get()` will just get None. And if you want to make it python way you should write:
`if person.get("wife"):`
which will make all detections of empty value and None.